libs/langgraph/langgraph/prebuilt/tool_node.py

```diff
@@ -221,8 +227,33 @@ def _func(
         config_list = get_config_list(config, len(tool_calls))
         with get_executor_for_config(config) as executor:
             outputs = [*executor.map(self._run_one, tool_calls, config_list)]
+        outputs, state_updates = zip(*outputs)
+        combined_state_updates = {}
+        for state_update in state_updates:
+            for k, v in state_update.items():
+                if k == self.messages_key:
+                    raise ValueError(
+                        "Cannot return state updates for the messages key."
+                    )
+
+                if k in combined_state_updates:
```

Comment:
Arguably should actually be allowed and follow the same semantics as the reducer. This would be solved by moving into a design where there's a node per tool. (In the context of the Tool Node, this might be the best that can be done.)